File: utio/models/Alert.js
/**
 * 报警记录模型
 * 对应表：alerts
 */

class Alert {
  constructor(data = {}) {
    this.id = data.id || null
    this.level = data.level || 'info'
    this.message = data.message || ''
    this.alertType = data.alert_type || data.alertType || null
    this.relatedData = data.related_data || data.relatedData || null
    this.isRead = data.is_read !== undefined ? data.is_read : (data.isRead !== undefined ? data.isRead : 0)
    this.createdAt = data.created_at || data.createdAt || null
  }

  /**
   * 转换为数据库字段格式
   */
  toDbFormat() {
    return {
      id: this.id,
      level: this.level,
      message: this.message,
      alert_type: this.alertType,
      related_data: this.relatedData ? JSON.stringify(this.relatedData) : null,
      is_read: this.isRead ? 1 : 0,
    }
  }

  /**
   * 转换为前端格式
   */
  toFrontendFormat() {
    return {
      id: this.id,
      level: this.level,
      message: this.message,
      alertType: this.alertType,
      relatedData: typeof this.relatedData === 'string' ? JSON.parse(this.relatedData) : this.relatedData,
      isRead: this.isRead,
      time: this.createdAt ? new Date(this.createdAt).toISOString() : null,
      createdAt: this.createdAt,
    }
  }

  /**
   * 验证数据有效性
   */
  validate() {
    const errors = []
    if (!['info', 'warning', 'error'].includes(this.level)) {
      errors.push('报警级别必须是 info、warning 或 error')
    }
    if (!this.message || this.message.trim() === '') {
      errors.push('报警消息不能为空')
    }
    return errors
  }
}

module.exports = Alert

File: utio/springboot/src/main/java/com/greenhouse/entity/Alert.java
package com.greenhouse.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

/**
 * 报警记录实体类
 * 对应表：alerts
 */
@Entity
@Table(name = "alerts", indexes = {
    @Index(name = "idx_level", columnList = "level"),
    @Index(name = "idx_alert_type", columnList = "alert_type"),
    @Index(name = "idx_is_read", columnList = "is_read"),
    @Index(name = "idx_created_at", columnList = "created_at")
})
@Data
@EntityListeners(AuditingEntityListener.class)
public class Alert {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 报警级别（info/warning/error）
     */
    @Column(nullable = false, length = 20)
    private String level;

    /**
     * 报警消息
     */
    @Column(nullable = false, length = 500)
    private String message;

    /**
     * 报警类型（temperature/soil_moisture/light/automation等）
     */
    @Column(name = "alert_type", length = 50)
    private String alertType;

    /**
     * 关联数据（JSON格式）
     */
    @Column(name = "related_data", columnDefinition = "JSON")
    private String relatedData;

    /**
     * 是否已读（0未读/1已读）
     */
    @Column(name = "is_read", nullable = false)
    private Boolean isRead;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

